TKMovesets2/GameSpecific/Structures/Structs_t8.h

```diff
@@ -66,31 +66,46 @@ namespace StructsT8
 
 	// -- Main data type -- //
 
-	struct PushbackExtradata //todo
+	struct PushbackExtradata 
 	{
 		uint16_t horizontal_offset;
 	};
 
-	struct Pushback //todo
+	struct Pushback
 	{
 		uint16_t duration;
 		uint16_t displacement;
-		uint32_t num_of_loops;
+		uint32_t offsets_count;
 		PushbackExtradata* extradata_addr;
 	};
 
-	struct Requirement //todo
+	struct Requirement
 	{
 		uint32_t condition;
-		union
+		struct
 		{
-			uint32_t param_unsigned;
-			int32_t param_signed;
-			float param_float;
+			uint32_t param1_unsigned;
+			uint32_t param2_unsigned;
+			uint32_t param3_unsigned;
+			uint32_t param4_unsigned;
+		};
+		struct
+		{
+			int32_t param1_signed;
+			int32_t param2_signed;
+			int32_t param3_signed;
+			int32_t param4_signed;
+		};
+		struct
+		{
+			float param1_float;
+			float param2_float;
+			float param3_float;
+			float param4_float;
 		};
```

Comment:
This whole param thing shoudl be an array of unions, that way each individual item can be accessed using an index followed by the type,
Like `req->param[0].floating_point;` `req->param[1].signed`